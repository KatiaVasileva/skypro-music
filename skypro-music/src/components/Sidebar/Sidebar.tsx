import Image from "next/image";
import playlist01 from "./img/playlist01.png";
import playlist02 from "./img/playlist02.png";
import playlist03 from "./img/playlist03.png";
import styles from "./Sidebar.module.css";
import Sprite from "@/public/Icon/IconSearch";

function Sidebar() {
  return (
    <div className={styles.sidebar}>
      <div className={styles.personal}>
        <p className={styles.personalName}>Sergey.Ivanov</p>
        <Sprite />
        <div className={styles.icon}>
          <svg>
            <use href="#logout"></use>
          </svg>
        </div>
      </div>
      <div className={styles.block}>
        <div className={styles.list}>
          <div className={styles.item}>
            <a className={styles.link} href="#">
              <Image
                className={styles.img}
                src={playlist01}
                alt="day's playlist"
                width={250}
                height={150}
              />
            </a>
          </div>
          <div className={styles.item}>
            <a className={styles.link} href="#">
              <Image
                className={styles.img}
                src={playlist02}
                alt="day's playlist"
                width={250}
                height={150}
              />
            </a>
          </div>
          <div className={styles.item}>
            <a className={styles.link} href="#">
              <Image
                className={styles.img}
                src={playlist03}
                alt="day's playlist"
                width={250}
                height={150}
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  );
}

export default Sidebar;
