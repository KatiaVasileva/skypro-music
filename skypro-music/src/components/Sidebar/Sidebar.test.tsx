import ReduxProvider from "@/store/ReduxProvider";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Sidebar from "./Sidebar";

describe("Sidebar component", () => {
  it("renders correctly", () => {
    const { container } = render(
      <ReduxProvider>
        <Sidebar />
      </ReduxProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
