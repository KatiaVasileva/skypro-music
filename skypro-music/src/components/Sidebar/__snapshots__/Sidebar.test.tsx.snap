// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component renders correctly 1`] = `
<div>
  <div
    class="sidebar"
  >
    <div
      class="personal"
    >
      <p
        class="personalName"
      />
      <div
        class="icon"
      >
        <svg>
          <use
            xlink:href="/img/icon/sprite.svg#logout"
          />
        </svg>
      </div>
    </div>
    <div
      class="block"
    >
      <div
        class="list"
      >
        <div
          class="item"
        >
          <a
            href="/playlist/selection/1"
          >
            <img
              alt="day's playlist"
              class="img"
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height="150"
              src="/_next/image?url=%2Fimg%2Fplaylist01.png&w=640&q=75"
              srcset="/_next/image?url=%2Fimg%2Fplaylist01.png&w=256&q=75 1x, /_next/image?url=%2Fimg%2Fplaylist01.png&w=640&q=75 2x"
              style="color: transparent;"
              width="250"
            />
          </a>
        </div>
        <div
          class="item"
        >
          <a
            href="/playlist/selection/2"
          >
            <img
              alt="day's playlist"
              class="img"
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height="150"
              src="/_next/image?url=%2Fimg%2Fplaylist02.png&w=640&q=75"
              srcset="/_next/image?url=%2Fimg%2Fplaylist02.png&w=256&q=75 1x, /_next/image?url=%2Fimg%2Fplaylist02.png&w=640&q=75 2x"
              style="color: transparent;"
              width="250"
            />
          </a>
        </div>
        <div
          class="item"
        >
          <a
            href="/playlist/selection/3"
          >
            <img
              alt="day's playlist"
              class="img"
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height="150"
              src="/_next/image?url=%2Fimg%2Fplaylist03.png&w=640&q=75"
              srcset="/_next/image?url=%2Fimg%2Fplaylist03.png&w=256&q=75 1x, /_next/image?url=%2Fimg%2Fplaylist03.png&w=640&q=75 2x"
              style="color: transparent;"
              width="250"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
